client/trino-cli/src/main/java/io/trino/cli/MarkdownTablePrinter.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.cli;
+
+import com.google.common.collect.ImmutableSet;
+import io.trino.client.Column;
+import net.steppschuh.markdowngenerator.table.Table;
+import net.steppschuh.markdowngenerator.table.TableRow;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.client.ClientStandardTypes.BIGINT;
+import static io.trino.client.ClientStandardTypes.DECIMAL;
+import static io.trino.client.ClientStandardTypes.DOUBLE;
+import static io.trino.client.ClientStandardTypes.INTEGER;
+import static io.trino.client.ClientStandardTypes.REAL;
+import static io.trino.client.ClientStandardTypes.SMALLINT;
+import static io.trino.client.ClientStandardTypes.TINYINT;
+import static java.lang.System.lineSeparator;
+import static java.util.Objects.requireNonNull;
+
+public class MarkdownTablePrinter
+        implements OutputPrinter
+{
+    private static final Set<String> NUMERIC_TYPES = ImmutableSet.of(TINYINT, SMALLINT, INTEGER, BIGINT, REAL, DOUBLE, DECIMAL);
+    private final List<String> fieldNames;
+    private final List<Integer> alignments;
+    private final Writer writer;
+
+    public MarkdownTablePrinter(List<Column> columns, Writer writer)
+    {
+        requireNonNull(columns, "columns is null");
+        this.fieldNames = columns.stream()
+                .map(Column::getName)
+                .collect(toImmutableList());
+        this.alignments = columns.stream()
+                .map(Column::getTypeSignature)
+                .map(signature -> NUMERIC_TYPES.contains(signature.getRawType()) ? Table.ALIGN_RIGHT : Table.ALIGN_LEFT)
+                .collect(toImmutableList());
+        this.writer = requireNonNull(writer, "writer is null");
+    }
+
+    @Override
+    public void printRows(List<List<?>> rows, boolean complete)
+            throws IOException
+    {
+        Table.Builder tableBuilder = new Table.Builder()
```

Comment:
tableBuilder could be moved to the constructor as well

Reply:
It needs to be reset for every batch of rows.

Reply:
why?

Reply:
It accumulates rows to be printed, and I don't see a method to reset it. If it was initialized in the constructor, instead of here, it would keep printing the same lines in every batch.

Reply:
when the number of rows is greater then OutputHandler.MAX_BUFFERED_ROWS multiple headers and table separator will be printed. I guess that we need to approach it other way.

Reply:
I dropped the dependency, copied the code from `AlignedTablePrinter` and simplified it.